ServerCore/Pages/Puzzles/FileManagement.cshtml.cs

```diff
@@ -287,6 +287,36 @@ private bool PuzzleExists(int puzzleId)
             return _context.Puzzles.Any(e => e.ID == puzzleId);
         }
 
+        /// <summary>
+        /// Promotes files named "index.html" to the CustomURL or CustomSolutionURL properties of the puzzle.
+        /// </summary>
+        /// <param name="file">The file being uploaded</param>
+        private void UpdateCustomURLs(ContentFile file)
+        {
+            // Skip this if:
+            // - the file is not named index.html
+            // - the file is not a material or solution file
+            // - the appropriate Custom[Solution]URL is non-blank
+            if (!file.ShortName.EndsWith("index.html") ||
```

Comment:
I'd suggest using Path.GetFileName for this. I also wonder if there's other relevant info about the path... does anything generate multiple files named index.html?